CONTRIBUTING.md

```diff
@@ -1,40 +1,44 @@
+    * Commits should be as narrow in scope as possible. Commits that span hundreds of lines across multiple unrelated functions and/or files are very hard for maintainers to follow. After about a week they'll probably be hard for you to follow too.
+    * Try to avoid making minor commits for fixing typos or linting errors. Since you've already set up a pre-commit hook to run `flake8` before a commit, you shouldn't be committing linting issues anyway.
+    * A more in-depth guide to writing great commit messages can be found in Chris Beam's [*How to Write a Git Commit Message*](https://chris.beams.io/posts/git-commit/)
+1. **Avoid frequent pushes to the main repository**. This goes for PRs opened against your fork as well. Our test build pipelines are triggered every time a push to the repository (or PR) is made. Try to batch your commits until you've finished working for that session, or you've reached a point where collaborators need your commits to continue their own work. This also provides you the opportunity to amend commits for minor changes rather than having to commit them on their own because you've already pushed.
+    * This includes merging master into your branch. Try to leave merging from master for after your PR passes review; a maintainer will bring your PR up to date before merging. Exceptions to this include: resolving merge conflicts, needing something that was pushed to master for your branch, or something was pushed to master that could potentionally affect the functionality of what you're writing.
+1. **Don't fight the framework**. Every framework has its flaws, but the frameworks we've picked out have been carefully chosen for their particular merits. If you can avoid it, please resist reimplementing swathes of framework logic - the work has already been done for you!
+1. If someone is working on a merge request, **do not open your own merge request for the same task**. Instead, collaborate with the author(s) of the existing merge request. Communication is key, and there's no point in two separate implementations of the same thing.
+    * One option is to fork the other contributor's repository and submit your changes to their branch with your own merge request. We suggest following these guidelines when interacting with their repository as well.
+1. **Work as a team** and collaborate whereever possible. Keep things friendly and help each other out - these are shared projects and nobody likes to have their feet trodden on.
+1. **Internal projects are internal**. As a contributor, you have access to information that the rest of the server does not. With this trust comes responsibility - do not release any information you have learned as a result of your contributor position. We are very strict about announcing things at specific times, and many staff members will not appreciate a disruption of the announcement schedule.
```

Comment:
you enumerate these as 1. 1. 1. 1. 1. 1.

![image](https://user-images.githubusercontent.com/22679924/51083859-390ca380-1721-11e9-8a37-f7f04d54b0de.png)

is that intentional, did you want to use - or 1. 2. 3. ?

Reply:
Intentional, look at the rendered markdown.

Reply:
Markdown, is made for humans to read and only coincidentally for computers to render. If it's supposed to be an unordered list, use `-` or `*`. If it's supposed to be ordered, enumerate it.